mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/ReconnectionStrategyTypeBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.declaration.type;
+
+import static org.mule.metadata.api.builder.BaseTypeBuilder.create;
+import static org.mule.metadata.api.model.MetadataFormat.JAVA;
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.builder.ObjectTypeBuilder;
+import org.mule.metadata.api.builder.TypeBuilder;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.extension.api.declaration.type.annotation.TypeAliasAnnotation;
+
+import java.util.Map;
+
+/**
+ * Creates instances of {@link MetadataType} which represent a reconnection strategy
+ *
+ * @since 1.0
+ */
+public final class ReconnectionStrategyTypeBuilder extends InfrastructureTypeBuilder {
+
+  /**
+   * @return a {@link MetadataType} representation of a retry policy
+   */
+  public MetadataType builReconnectionStrategyType() {
+    BaseTypeBuilder typeBuilder = create(JAVA);
```

Comment:
every type that is parsed using the `ExtensionObjectTypeHandler` is enriched with an `TypeAliasAnnotation`. To be consistent with the other types, we could add the Alias to hand-built types like this one (in this case it could be `ReconnectionStrategy`)